doc/CMakeLists.txt

```diff
@@ -66,8 +66,10 @@ endif()
 find_program(TEXI2PDF texi2pdf
              ${CYGWIN_INSTALL_PATH}/bin /bin /usr/bin /usr/local/bin /sbin)
 mark_as_advanced(TEXI2PDF)
+find_program(PDFLATEX pdflatex
+             ${CYGWIN_INSTALL_PATH}/bin /bin /usr/bin /usr/local/bin /sbin)
 
-if (TEXI2PDF)
+if (TEXI2PDF AND PDFLATEX)
```

Comment:
Not sure if this is what we need.  To be clear, here is how these things SHOULD work: There should be a CMake option (looks like TEXI2PDF) that turns on generation of documentation (or a particular subset of documentation).  If that option is not set, then no docs are generated.  If the option IS set, then docs will be generated.  If the option is set and the required programs for generating docs do not exist (`pdflatex`), then CMake should generate an error.  It should NOT "automagically" just fail to generate the docs.  That will cause people unending grief ("I set TEXI2PDF but docs weren't generated"), and is against best practices.

Reply:
To be clear, I totally agree. This is a stop gap to allow compilation on travis MacOs which does not have latex and for which I do not know just like that how to install latex (+ the numerous latex packages which are needed). And I do not want to stop all documentation generation on MacOs just because of that (which would have been easier). Hence this hack.
I want to avoid adding too many documentation options, but I also want documentation building to be better tested.

I think we can revisit the problem later and just open an issue asking an enhancement for the time being.